lib/shim/transaction-shim.js

```diff
@@ -254,33 +254,41 @@ function setTransactionName(name) {
  *  The transport type that brought the headers. Usually `HTTP` or `HTTPS`.
  */
 function handleCATHeaders(headers, segment, transportType) {
-  // Is CAT enabled?
-  if (!this.agent.config.cross_application_tracer.enabled) {
-    this.logger.trace('CAT disabled, not extracting header.')
+  // TODO: rename function or replace functionality when CAT fully removed.
+
+  if (!headers) {
+    this.logger.debug('No headers for CAT or DT processing.')
     return
   }
-  if (!this.agent.config.encoding_key) {
-    this.logger.warn('Missing encoding key, not extracting CAT headers!')
-    return
-  } else if (!headers) {
-    this.logger.debug('No headers to search for CAT within')
+
+  const config = this.agent.config
+
+  if (!config.cross_application_tracer.enabled && !config.distributed_tracing.enabled) {
+    this.logger.trace('CAT and DT disabled, not extracting headers.')
     return
   }
 
   // Check that we're in an active transaction.
-  segment = segment || this.getSegment()
-  if (!segment || !segment.transaction.isActive()) {
-    this.logger.trace('Not adding CAT reply header, not in an active transaction.')
+  const currentSegment = segment || this.getSegment()
+  if (!currentSegment || !currentSegment.transaction.isActive()) {
+    this.logger.trace('Not processing headers for CAT or DT, not in an active transaction.')
     return
   }
```

Comment:
Didn't love re-assigning of the incoming function argument as that can sometimes lead to future bugs when trying to check against the original value if you don't notice the reassignment. While the risk was low here, wanted to clean up the pattern.

Reply:
Nice clean up.